frontend/src/metabase-lib/expressions/suggest.ts

```diff
@@ -149,49 +154,66 @@ export function suggest({
 
   if (_.last(matchPrefix) !== "]") {
     suggestions.push(
-      ...legacyQuery
-        .dimensionOptions(() => true)
-        .all()
-        .map(dimension => ({
-          type: "fields",
-          name: getDimensionName(dimension),
-          text: formatDimensionName(dimension) + " ",
-          alternates: EDITOR_FK_SYMBOLS.symbols.map(symbol =>
-            getDimensionName(dimension, symbol),
-          ),
-          index: targetOffset,
-          icon: dimension.icon(),
-          order: 2,
-          dimension,
-        })),
+      ...Lib.expressionableColumns(query, stageIndex, targetOffset).map(
+        column => {
+          const displayInfo = Lib.displayInfo(query, stageIndex, column);
+          return {
+            type: "fields",
+            name: displayInfo.longDisplayName,
+            text: formatIdentifier(displayInfo.displayName) + " ",
+            alternates: EDITOR_FK_SYMBOLS.symbols.map(symbol =>
+              getDisplayNameWithSeparator(displayInfo.longDisplayName, symbol),
+            ),
+            index: targetOffset,
+            icon: getColumnIcon(column),
+            order: 2,
+            ...column,
+          };
+        },
+      ),
     );
 
-    const segments = legacyQuery.table()?.segments;
+    const segments = Lib.availableSegments(query, stageIndex);
+
     if (segments) {
       suggestions.push(
-        ...segments.map(segment => ({
-          type: "segments",
-          name: segment.name,
-          text: formatSegmentName(segment),
-          index: targetOffset,
-          icon: "segment",
-          order: 3,
-        })),
+        ...segments.map(segment => {
+          // TODO: fix type
+          const displayInfo = Lib.displayInfo(
+            query,
+            stageIndex,
+            segment as any,
```

Comment:
displayInfo returns correct value, so probably we need to support this type in displayInfo?